src/platform/packages/shared/kbn-coloring/src/shared_components/color_mapping/color/color_handling.test.ts

```diff
@@ -143,7 +147,7 @@ describe('Color mapping - color generation', () => {
         assignments: [
           {
             color: { type: 'colorCode', colorCode: 'red' },
-            rule: { type: 'matchExactly', values: ['configuredAssignment'] },
+            rules: [{ type: 'raw', value: ['configuredAssignment'] }],
```

Comment:
This is interesting: the value is now typed as `SerializedValue`, but here it was configured as an array of string. Lukly the `String(rawValue)` that we apply to compute the key in colorFactory returns just the string of the array element and both things matches, but below we are testing for just the string value and not it's array version as we should probably do expect(colorFactory(['configuredAssignment'])).toBe('red');

I'm wondering if we have considered arrays like that in our code or we just strinfigy them as their serialized form

Reply:
Yeah true, but this is the negative in using the `unknown` (i.e. `SerializedValue) type.

As far as the question about handling arrays. The previous logic split **_only_** `MultiFieldKey` values into array of values, nothing else. So in terms of migration, arrays are only `MultiFieldKey` values.

But in general, I think you are asking about if the `RawValue` and thus the `SerializedValue` can be an array. I did not consider this thinking that the only way that would occur is with the **Last value** (as array) aggregation which is only supported in the `Metric` and `Table for purposes an both disallow coloring based on array values.

But I didn't think about if the value itself was mapped as an array.

Reply:
>Last value (as array) aggregation which is only supported in the Metric and Table and both disallow coloring based on array values.

Correct, I believe with that we are "safe" from that case.
I've tested locally and you are right, Elastisearch doesn't return array of values in the Term agg,  but threat them as "duplicated" documents.